In [2]:
import pandas as pd

In [3]:
#Load original dataset
raw_data= pd.read_csv('/content/heart_cleveland_upload.csv')

#Shows all the info of dataset
print('Information of raw')
raw_data.info()

#Shows statistical info.
raw_data.describe()

Information of raw
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 297 entries, 0 to 296
Data columns (total 14 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   age        297 non-null    int64  
 1   sex        297 non-null    int64  
 2   cp         297 non-null    int64  
 3   trestbps   297 non-null    int64  
 4   chol       297 non-null    int64  
 5   fbs        297 non-null    int64  
 6   restecg    297 non-null    int64  
 7   thalach    297 non-null    int64  
 8   exang      297 non-null    int64  
 9   oldpeak    297 non-null    float64
 10  slope      297 non-null    int64  
 11  ca         297 non-null    int64  
 12  thal       297 non-null    int64  
 13  condition  297 non-null    int64  
dtypes: float64(1), int64(13)
memory usage: 32.6 KB


age         sex          cp    trestbps        chol         fbs  \
count  297.000000  297.000000  297.000000  297.000000  297.000000  297.000000   
mean    54.542088    0.676768    2.158249  131.693603  247.350168    0.144781   
std      9.049736    0.468500    0.964859   17.762806   51.997583    0.352474   
min     29.000000    0.000000    0.000000   94.000000  126.000000    0.000000   
25%     48.000000    0.000000    2.000000  120.000000  211.000000    0.000000   
50%     56.000000    1.000000    2.000000  130.000000  243.000000    0.000000   
75%     61.000000    1.000000    3.000000  140.000000  276.000000    0.000000   
max     77.000000    1.000000    3.000000  200.000000  564.000000    1.000000   

          restecg     thalach       exang     oldpeak       slope          ca  \
count  297.000000  297.000000  297.000000  297.000000  297.000000  297.000000   
mean     0.996633  149.599327    0.326599    1.055556    0.602694    0.676768   
std      0.994914   22.941562    0.469761    1.166123    0.618187    0.938965   
min      0.000000   71.000000    0.000000    0.000000    0.000000    0.000000   
25%      0.000000  133.000000    0.000000    0.000000    0.000000    0.000000   
50%      1.000000  153.000000    0.000000    0.800000    1.000000    0.000000   
75%      2.000000  166.000000    1.000000    1.600000    1.000000    1.000000   
max      2.000000  202.000000    1.000000    6.200000    2.000000    3.000000   

             thal   condition  
count  297.000000  297.000000  
mean     0.835017    0.461279  
std      0.956690    0.499340  
min      0.000000    0.000000  
25%      0.000000    0.000000  
50%      0.000000    0.000000  
75%      2.000000    1.000000  
max      2.000000    1.000000

In [4]:
#Displaying top 5 rows of raw_data
print('Original dataset')
raw_data.head(5)

Original dataset


age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   69    1   0       160   234    1        2      131      0      0.1      1   
1   69    0   0       140   239    0        0      151      0      1.8      0   
2   66    0   0       150   226    0        0      114      0      2.6      2   
3   65    1   0       138   282    1        2      174      0      1.4      1   
4   64    1   0       110   211    0        2      144      1      1.8      1   

   ca  thal  condition  
0   1     0          0  
1   2     0          0  
2   0     0          0  
3   1     0          1  
4   0     0          0

In [5]:
##Renaming technical column names to meaningful labels
raw_data.rename(columns={'cp': 'chest pain type', 'trestbps':'resting blood pressure','chol':'serum cholestoral(mg/dl)','fbs':'fasting blood sugar > 120 mg/dl',
                         'restecg':'resting electrocardiographic results','thalach':'maximum heart rate achieved','exang':'exercise induced angina','oldpeak':'Induced depression(exercise vs rest)',
                         'slope':'slope peak exercise','ca':'colored vessels','thal':'thalassemia','condition':'disease condition'}, inplace=True)
display(raw_data.head(2))


age  sex  chest pain type  resting blood pressure  \
0   69    1                0                     160   
1   69    0                0                     140   

   serum cholestoral(mg/dl)  fasting blood sugar > 120 mg/dl  \
0                       234                                1   
1                       239                                0   

   resting electrocardiographic results  maximum heart rate achieved  \
0                                     2                          131   
1                                     0                          151   

   exercise induced angina  Induced depression(exercise vs rest)  \
0                        0                                   0.1   
1                        0                                   1.8   

   slope peak exercise  colored vessels  thalassemia  disease condition  
0                    1                1            0                  0  
1                    0                2            0                  0

In [6]:
#Check for missing values
raw_data.isnull().sum()

age                                     0
sex                                     0
chest pain type                         0
resting blood pressure                  0
serum cholestoral(mg/dl)                0
fasting blood sugar > 120 mg/dl         0
resting electrocardiographic results    0
maximum heart rate achieved             0
exercise induced angina                 0
Induced depression(exercise vs rest)    0
slope peak exercise                     0
colored vessels                         0
thalassemia                             0
disease condition                       0
dtype: int64

In [7]:
#Check for duplicates in dataset.
print(raw_data.duplicated().sum())

0


*   Dataset has 14 columns and 297 entries.
*   There are no missing values.
*   There are no duplicate

In [8]:
#Replacing/converting categorical variables to interpretable form.
raw_data.replace({'sex':{1:'male', 0:'female'},'chest pain type':{ 0: 'typical angina',1: 'atypical angina',2: 'non-anginal pain',3: 'asymptomatic'},
                  'fasting blood sugar > 120 mg/dl':{1:'true', 0:'false'},
                  'resting electrocardiographic results':{0:'normal',1:'wave abnormality',2:'left ventricular hypertrophy'},
                  'exercise induced angina':{1:'yes', 0:'no'},'slope peak exercise':{0:'upsloping',1:'flat',2:'downsloping'},'thalassemia':{0:'normal', 1:'fixed defect', 2:'reversable defect'}
                  ,'disease condition':{0:'no disease', 1:'disease'}},inplace=True)
display(raw_data.head(3))

age     sex chest pain type  resting blood pressure  \
0   69    male  typical angina                     160   
1   69  female  typical angina                     140   
2   66  female  typical angina                     150   

   serum cholestoral(mg/dl) fasting blood sugar > 120 mg/dl  \
0                       234                            true   
1                       239                           false   
2                       226                           false   

  resting electrocardiographic results  maximum heart rate achieved  \
0         left ventricular hypertrophy                          131   
1                               normal                          151   
2                               normal                          114   

  exercise induced angina  Induced depression(exercise vs rest)  \
0                      no                                   0.1   
1                      no                                   1.8   
2                      no                                   2.6   

  slope peak exercise  colored vessels thalassemia disease condition  
0                flat                1      normal        no disease  
1           upsloping                2      normal        no disease  
2         downsloping                0      normal        no disease

In [9]:
#Export the cleaned dataset
raw_data.to_csv('cleaned_data_heart.csv',index=False)
print('Exported successfully')

Exported successfully
